lib/features/offline_mode/manager/opened_email_cache_manager.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:tmail_ui_user/features/offline_mode/extensions/list_detailed_email_hive_cache_extension.dart';
+import 'package:tmail_ui_user/features/offline_mode/model/detailed_email_hive_cache.dart';
+
+class OpenedEmailCacheManager {
+
+  final OpenedEmailHiveCacheClient _cacheClient;
+  final FileUtils _fileUtils;
+
+  OpenedEmailCacheManager(this._cacheClient, this._fileUtils);
+
+  Future<void> insertDetailedEmail(
+      AccountId accountId,
+      UserName userName,
+      DetailedEmailHiveCache detailedEmailCache
+  ) {
+    final keyCache = TupleKey(detailedEmailCache.emailId, accountId.asString, userName.value).encodeKey;
+    log('OpenedEmailCacheManager::insertDetailedEmail(): $keyCache');
+    return _cacheClient.insertItem(keyCache, detailedEmailCache);
+  }
+
+  Future<void> removeDetailedEmail(
+      AccountId accountId,
+      UserName userName,
+      String emailId
+  ) {
+    final keyCache = TupleKey(emailId, accountId.asString, userName.value).encodeKey;
+    log('OpenedEmailCacheManager::removeDetailedEmail(): $keyCache');
+    return _cacheClient.deleteItem(keyCache);
+  }
+
+  Future<List<DetailedEmailHiveCache>> getAllDetailedEmails(AccountId accountId, UserName userName) async {
+    final detailedEmailCacheList = await _cacheClient.getListByTupleKey(accountId.asString, userName.value);
+    detailedEmailCacheList.sortByLatestTime();
+    log('OpenedEmailCacheManager::getAllDetailedEmails():SIZE: ${detailedEmailCacheList.length}');
+    return detailedEmailCacheList;
+  }
+
+  Future<void> storeOpenedEmail(
+      AccountId accountId,
+      UserName userName,
+      DetailedEmail detailedEmail
+  ) async {
+    final listDetailedEmails = await getAllDetailedEmails(accountId, userName);
+
+    if (listDetailedEmails.length >= CachingConstants.maxNumberOpenedEmailsForOffline) {
+      final latestEmail = listDetailedEmails.last;
+      log('OpenedEmailCacheManager::handleStoreDetailedEmail():latestEmail: $latestEmail');
+      await removeDetailedEmail(accountId, userName, latestEmail.emailId);
+    }
+    await insertDetailedEmail(accountId, userName, detailedEmail.toHiveCache());
+  }
+
+  Future<bool> isOpenedDetailEmailCached(
+      AccountId accountId,
+      UserName userName,
+      DetailedEmail detailedEmail
+  ) async {
+    final emailContentPathExists = await _isFileExisted(detailedEmail);
+    final detailedEmailCacheExists = await _getDetailedEmailCache(accountId, userName, detailedEmail);
+
+    return emailContentPathExists == true && detailedEmailCacheExists;
+  }
+
+  Future<bool?> _isFileExisted(DetailedEmail detailedEmail) async {
+    final fileSaved = await _fileUtils.isFileExisted(
+      nameFile: detailedEmail.emailId.asString,
+      folderPath: detailedEmail.folderPath,
+    );
+    log('OpenedEmailCacheManager::_getDetailedEmailCache():_getEmailContentPath: $fileSaved');
+    return fileSaved;
+  }
+
+  Future<bool> _getDetailedEmailCache(
```

Comment:
Why a get function return a bool value?

Reply:
This func is used to check if the email is cached or not.

I'll change func name `_isDetailEmailExistedInCache`, it's clearly